### ISSUE TITLE
Create Python environment and lock dependencies

### ISSUE THREAD

**Issue by DakotaIrsik** [DakotaIrsik/TalkSmith#4]
Create Python 3.10/3.11 env and pin versions for torch, torchaudio, faster-whisper, whisperx, pyannote-audio, numpy, soundfile, rich, tqdm, pydantic, ffmpeg-python.

Acceptance Criteria:
- [ ] environment.yml (conda) and requirements.txt (venv) present
- [ ] scripts/make_env.(ps1|sh) to create/activate env across OSes
- [ ] python -c "import torch; print(torch.cuda.is_available())" returns True on GPU boxes

Labels: Area/Infra, Type/chore, OS/Windows, OS/Linux, OS/macOS
Priority: P0
Depends on: Install FFmpeg and add to PATH

**Comment by DakotaIrsik** [comment]
✅ **Issue Resolved**

All acceptance criteria have been completed:

- [x] **environment.yml (conda)** - Complete Conda environment specification with PyTorch, CUDA, and all dependencies
- [x] **requirements.txt (venv)** - Comprehensive pip requirements with version pinning for torch, torchaudio, faster-whisper, whisperx, pyannote-audio, and all utilities
- [x] **scripts/make_env.ps1** - Windows PowerShell script for automated environment creation with venv or conda
- [x] **scripts/make_env.sh** - Linux/macOS bash script for automated environment creation
- [x] **CUDA verification** - Scripts verify torch.cuda.is_available() returns True on GPU systems

Created comprehensive environment setup:

**Dependencies included:**
- Core: PyTorch 2.1.0+, torchaudio
- Transcription: faster-whisper, whisperx
- Diarization: pyannote.audio, resemblyzer
- Audio: soundfile, librosa, noisereduce, ffmpeg-python
- Utilities: numpy, scipy, pandas, tqdm, rich
- Configuration: pydantic, pydantic-settings
- Export: webvtt-py, srt
- Optional: Google API clients for cloud sync
- Testing: pytest, pytest-cov, pytest-asyncio, pytest-mock
- Development: black, isort, flake8, mypy

**Setup scripts support:**
- Both venv and conda environments
- Windows (PowerShell) and Linux/macOS (bash)
- Automatic CUDA version selection (cu118, cu121, or cpu)
- Verification of Python version and CUDA availability
- Clear step-by-step output and next steps guidance

**Usage:**
```bash
# Windows
.\scripts\make_env.ps1

# Linux/macOS  
./scripts/make_env.sh

# With conda
.\scripts\make_env.ps1 -EnvType conda
```

Environment setup is complete and ready for use!
